backend/app/utils/rate_limiter.py

```
@@ -0,0 +1,154 @@
+    GET_PROFILE = ["60/minute", "1000/hour"]
+    UPDATE_PROFILE = ["30/minute", "500/hour"]
+    
+    # Lenient limits for read-only operations
+    GET_TEMPLATES = ["100/minute"]
+    HEALTH_CHECK = ["200/minute"]
+    
+    # Upload limits (file processing)
+    UPLOAD_RESUME = ["10/minute", "50/hour"]
+
+
+def rate_limit_exceeded_handler(request: Request, exc: RateLimitExceeded):
```

Ruff (0.14.14)

[warning] Unused function argument: `exc`

(ARG001)